spock-core/src/main/java/spock/util/environment/Jvm.java

```diff
@@ -128,30 +101,18 @@ public boolean isJava12() {
   }
 
   /**
-   * Tells whether the Java version is compatible with Java 5.
+   * Tells whether the Java version is equal to the given major Java version.
    *
-   * @return whether the Java version is compatible with Java 5
-   */
-  public boolean isJava5Compatible() {
-    return javaSpecVersionNumber.getMajor() > 1 || javaSpecVersionNumber.getMinor() >= 5;
-  }
-
-  /**
-   * Tells whether the Java version is compatible with Java 6.
-   *
-   * @return whether the Java version is compatible with Java 6
-   */
-  public boolean isJava6Compatible() {
-    return javaSpecVersionNumber.getMajor() > 1 || javaSpecVersionNumber.getMinor() >= 6;
-  }
-
-  /**
-   * Tells whether the Java version is compatible with Java 7.
-   *
-   * @return whether the Java version is compatible with Java 7
+   * @since 2.0
+   * @param majorJavaVersion major java version (e.g. 8, 12, 17) to check the Java version is equal to
+   * @return whether the Java version is equal to the given major Java version
    */
-  public boolean isJava7Compatible() {
-    return javaSpecVersionNumber.getMajor() > 1 || javaSpecVersionNumber.getMinor() >= 7;
+  public boolean isJavaVersion(int majorJavaVersion) {
```

Comment:
I still think we should add the simple accessor methods, maybe up to java 20?

Reply:
Ok. Maybe up to Java 17 - the next LTS? We can't be sure how will be the versioning scheme look like after that :).